### PR TITLE
Update documentation

### LINKED ISSUE #1669: Improve documentation for package maintainers
- add section about uninstallation
- add section about protocol registration (requires #1668)
- add section about `.desktop` files

### PR DESCRIPTION
**Pull request by erri120**
Resolves #1669.

### CODE REVIEW

File: docs/developers/Contributing.md [hunk 1]
```diff
@@ -20,13 +20,21 @@ Translations are currently handled via the IDE. See [this issue](https://github.
...
 - Set `INSTALLATION_METHOD_PACKAGE_MANAGER` when publishing. We have an integrated updater that will notify users about new versions. If you set this flag, we'll tell the user to update the App with their package manager.
-- Let us know if you have questions or if you published a new package by joining our [Discord](https://discord.gg/ReWTxb93jS).
 
 We publish the App using [PupNet](https://github.com/kuiperzone/PupNet-Deploy). Releases are built using GitHub Actions, see [`build-linux-pupnet.yaml`](https://github.com/Nexus-Mods/NexusMods.App/blob/main/.github/workflows/build-linux-pupnet.yaml) for details. PupNet will use `dotnet publish` before packaging the result in some specialized format.
 
 If you don't wish to use PupNet, you should still use `dotnet publish` over `dotnet build`. The arguments we use with `dotnet publish` can be found in [`app.pupnet.conf`](https://github.com/Nexus-Mods/NexusMods.App/blob/main/src/NexusMods.App/app.pupnet.conf). Also see the [`dotnet publish` documentation](https://learn.microsoft.com/en-us/dotnet/core/tools/dotnet-publish#options) for a list of options.
```
Comment (MattSturgeon) [inline docs/developers/Contributing.md:31]:
```suggestion
If you don't wish to use PupNet, you should still prefer `dotnet publish` over `dotnet build`. The arguments we use with `dotnet publish` can be found in [`app.pupnet.conf`](https://github.com/Nexus-Mods/NexusMods.App/blob/main/src/NexusMods.App/app.pupnet.conf). Also see the [`dotnet publish` documentation](https://learn.microsoft.com/en-us/dotnet/core/tools/dotnet-publish#options) for a list of options.
```

Some distros will have technical reasons to use `dotnet build`, so perhaps this wording should be less strong?

File: docs/developers/Contributing.md [hunk 2]
```diff
@@ -20,13 +20,21 @@ Translations are currently handled via the IDE. See [this issue](https://github.
 
 ## For Package Maintainers
 
-If you want to create a package for your distribution, here are some helpful tips to get started:
+If you want to create a package for your distribution, here are some helpful tips to get started. If you have questions or want to let us know about your new package, consider joining our [Discord](https://discord.gg/ReWTxb93jS).
 
-- If possible, use `nexusmods-app` for the package name.
+- If possible, use `nexusmods-app` for the package name and `com.nexusmods.app` as the rDNS ID.
 - We ship a build of `7zz` and use that executable unless you set `NEXUSMODS_APP_USE_SYSTEM_EXTRACTOR=1` when publishing. See [this issue](https://github.com/Nexus-Mods/NexusMods.App/issues/1306#issuecomment-2095755699) for details.
 - Set `INSTALLATION_METHOD_PACKAGE_MANAGER` when publishing. We have an integrated updater that will notify users about new versions. If you set this flag, we'll tell the user to update the App with their package manager.
```
Comment (MattSturgeon) [inline docs/developers/Contributing.md:27]: These bullet points do not make it clear that these are build props, or how they should be set.

At first glance many unix-y people would assume these are environment variables.

File: docs/users/SystemRequirements.md [hunk 3]
```diff
@@ -31,10 +31,12 @@ This page lists the recommended system requirements to install and run the app o
 <sub>\* Storage space required does not factor in the number of games managed, the size of the managed games and the mods installed.</sub> -->
 
 ### Linux Dependencies
+
 The App requires the following programs to be available at runtime:
```
Comment (MattSturgeon) [inline docs/users/SystemRequirements.md:35]:
```suggestion
The App requires the following programs to be available on the system `PATH` at runtime:
```

**Review by MattSturgeon — COMMENTED** [review]
These changes look good, but reading through the docs I've commented on some other lines that could do with some attention IMO.

**Comment by erri120** [comment]
> These changes look good, but reading through the docs I've commented on some other lines that could do with some attention IMO.

Thanks, will take a look.

File: docs/users/SystemRequirements.md (hunk 3 above)
Comment (Pickysaurus) [reply docs/users/SystemRequirements.md:35]: I'll add this to my next commit of the user docs @erri120 FYI

**Review by Al12rs — APPROVED** [review]

**Pull request merged**